en/03_Upgrading/07_Deprecations.md

```diff
@@ -4,27 +4,49 @@ title: Deprecations
 
 # Deprecations
 
-## Updating your codebase to use avoid using deprecated API in Silverstripe CMS
-
 You'll need to go through your codebase to remove references to deprecated APIs and update your project logic.
 
 Major releases of Silverstripe CMS introduce many API changes. They include a deprecation message that will either tell you:
 - to simply use a different yet equivalent API instead, or
 - the API in question has no replacement. This is usually for API that was mostly irrelevant.
 
+## Enabling deprecation warnings
+
 To see deprecation warnings in your project and in your project's CI, add `SS_DEPRECATION_ENABLED=true` to your project's `.env` file.
 
-Alternatively, add the following line to you project's `app/_config.php`.
+Alternatively, add the following line to your project's `app/_config.php`.
 ```php
 Deprecation::enable()
 ```
 
-Deprecation notices will only ever show if your `SS_ENVIRONMENT_TYPE` is set to `dev`.
+Deprecation warnings will only ever show if your `SS_ENVIRONMENT_TYPE` is set to `dev`.
+
+Your project will now start emitting deprecation warnings into the error logger. These warnings will not appear in your browser or CLI output - you will need to make sure you have a logging handler attached to the default `Psr\Log\LoggerInterface` or `Psr\Log\LoggerInterface.errorhandler` singletons. For example, to log to a file you can add this to your yml configuration:
+
+```yml
+SilverStripe\Core\Injector\Injector:
+  ErrorLogFileHandler:
+    class: Monolog\Handler\StreamHandler
+    constructor:
+      - "../silverstripe.log" # an absolute path, or a path relative to the index.php file (usually inside the public/ directory)
+      - "warning" # warning is the level deprecation warnings are logged as
+  Psr\Log\LoggerInterface.errorhandler:
+    calls:
+      ErrorLogFileHandler: [ pushHandler, [ '%$ErrorLogFileHandler' ] ]
+```
+
+See [Configuring error logging](/developer_guides/debugging/error_handling/#configuring-error-logging) to learn about other ways you can handle error logs.
+
+Additionally, if you want to view the warnings in your browser, you can install [lekoala/silverstripe-debugbar](https://github.com/lekoala/silverstripe-debugbar) as a dev dependency. Deprecation warnings will be logged in the "messages" tab of the debugbar. Be aware however that it will _not_ show you any deprecation warnings that are triggered from XHR requests or which are triggered after the middleware has finished generating the debugbar for the response.
```

Comment:
Given this it's only showing a subset of deprecation warnings, it seems like using debugbar here is actually a liability and not something we should encourage anyone to do.  It will give people doing upgrades a false sense of security that they fixed everything, when they haven't.

We're better off simply saying "you have to look at the log file which is the single source of truth" rather than giving only half working implementations.

Reply:
@maxime-rainville thoughts?

Reply:
I would create a sub header that says "Showing deprecation warning in your browser". I would put the "Be aware however ..." bit in a warning block to minimise the risk that people miss it.

Reply:
Done - the warning is in a warning block.